src/vm/wren_compiler.c

```diff
@@ -2695,6 +2727,92 @@ void expression(Compiler* compiler)
   parsePrecedence(compiler, PREC_LOWEST);
 }
 
+//Compound assigment handling
+TokenType compoundAssignmentStart(Compiler* compiler)
+{
+  //Consume the compound assign token
+  nextToken(compiler->parser);
+  switch(compiler->parser->previous.type)
+  {
+    //For the case of a normal = we handle the RHS
+    case TOKEN_EQ: {
+      expression(compiler);
+      return TOKEN_EQ;
+    } break;
+
+    case TOKEN_PLUSEQ:  return TOKEN_PLUS;
+    case TOKEN_MINUSEQ: return TOKEN_MINUS;
+    case TOKEN_STAREQ:  return TOKEN_STAR;
+    case TOKEN_SLASHEQ: return TOKEN_SLASH;
+    default: break;
+  };
+
+  UNREACHABLE();
+  return TOKEN_ERROR;
+}
+
+static void variableAssignment(Compiler* compiler, Variable variable)
+{
+  TokenType infixToken = compoundAssignmentStart(compiler);
+  if(infixToken == TOKEN_EQ) return;
+
+  //We're doing somevar = somevar + ...
+  //so we load the variable again as the arg to the infix
+  loadVariable(compiler, variable);
+  infixOpWithRule(compiler, getRule(infixToken));
+}
+
+static void fieldAssignment(Compiler* compiler, int field, bool insideMethod)
+{
+  TokenType infixToken = compoundAssignmentStart(compiler);
+  if(infixToken == TOKEN_EQ) return;
+
+  //We're doing _field = _field + ...
+  //so we load the field as the arg for the infix
+  if(insideMethod) {
+    emitByteArg(compiler, CODE_LOAD_FIELD_THIS, field);
+  } else {
+    loadThis(compiler);
+    emitByteArg(compiler, CODE_LOAD_FIELD, field);
+  }
+  //Then call the op which handles the RHS
+  infixOpWithRule(compiler, getRule(infixToken));
+}
+
+static void setterAssignment(Compiler* compiler, Code instruction, Signature signature)
+{
+  TokenType infixToken = compoundAssignmentStart(compiler);
+  if(infixToken == TOKEN_EQ) return;
+
+  //We're doing self.setter = self.getter + ...
+  //so to call the getter, we duplicate the self on the stack
+  emitOp(compiler, CODE_LOAD_LOCAL_0);
```

Comment:
This fails when 0 is not self (i.e inside a Fn closure). It should basically be stack -1, will revise.